### ISSUE TITLE
Unexpected issue with async/await

### ISSUE THREAD

**Issue by Ledslash** [eslint/eslint#11787]
**Tell us about your environment**

* **ESLint Version:**5.16.0
* **Node Version:**11.15.0
* **npm Version:**6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```js
{
    "env": {
        "es6": true,
        "node": true
    },
    "extends": ["eslint:recommended","prettier","plugin:prettier/recommended"],
    "parser": "esprima",
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "plugins": ["prettier"],
    "parserOptions": {
        "ecmaVersion": 8,
        "sourceType": "module"
    },
    "rules": {
	"prettier/prettier": "error"
    }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export default function makeFindReservationInCrs({ reservationCrs }) {
  return async function findReservationInCrs({ lastName, pnr } = {}) {
    if (!lastName) {
      throw new Error("You must supply any Last Name of a passenger");
    }
    if (!pnr) {
      throw new Error("You must supply a PNR");
    }
    const reservation = reservationCrs({ lastName, pnr });
    return reservation;
  };
}

```

**What did you expect to happen?**
eslint passes

**What actually happened? Please include the actual, raw output from ESLint.**
Parsing error: Unexpected token function

I've already checked the other issues with the same Async/Await problem but even tho i took their fix, i still have the same issue.

**Comment by kaicataldo** [comment]
Hi, is there a reason you have `esprima` listed as the parser? ESLint's default parser is `espree` (and does not need to be specified, since it's the default).

**Comment by g-plane** [comment]
I cannot reproduce this: https://astexplorer.net/#/gist/f10ff395cdb48b1b540852cca41dee39/61873a6977b14b5e5c1d0c0fb8c007844702b511

**Comment by Ledslash** [comment]
> Hi, is there a reason you have `esprima` listed as the parser? ESLint's default parser is `espree` (and does not need to be specified, since it's the default).

Im really not sure how that got in there, must have been a mistake i took somewhere. But after i changed it to espree it worked. Im sorry for the bother and than you
